src/fs.h

```diff
@@ -51,12 +51,26 @@ class path : public std::filesystem::path
     // Disallow std::string conversion method to avoid locale-dependent encoding on windows.
     std::string string() const = delete;
 
+    std::string u8string() const
+    {
+        auto utf8_str{std::filesystem::path::u8string()};
+        // utf8_str might either be std::string (C++17) or std::u8string
+        // (C++20). Convert both to std::string. This method can be removed
+        // after switching to C++20.
+        return *(std::string*)(&utf8_str);
```

Comment:
Is this well-defined behaviour?

Reply:
Practically probably yes. Theoretically no. I've pushed an alternative.